nre/bpf/nodefw/nodefw-ebpf/src/main.rs

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+#![no_std]
+#![no_main]
+
+use aya_bpf::{
+    bindings::xdp_action,
+    helpers::gen::bpf_ktime_get_ns,
+    macros::{map, xdp},
+    maps::HashMap,
+    programs::XdpContext,
+};
+use aya_log_ebpf::info;
+use core::mem;
+// TODO see if this is preferred over ptr_at
+// use memoffset::offset_of;
+use network_types::{
+    eth::{EthHdr, EtherType},
+    ip::{IpProto, Ipv4Hdr, Ipv6Hdr},
+    tcp::TcpHdr,
+    udp::UdpHdr,
+};
+use nodefw_common::Rule;
+
+const MAX_BLOCKLIST_SIZE: u32 = 1024;
+
+// the key is an ipv4 or ipv6 octet value expressed as an array.
+#[map]
+static BLOCKLIST: HashMap<[u8; 16usize], Rule> = HashMap::with_max_entries(MAX_BLOCKLIST_SIZE, 0);
+
+fn block_ip(ctx: &XdpContext, address: [u8; 16usize], dest_port: u16) -> bool {
+    unsafe {
+        // TODO find a way to check map len, if possible
+        // if BLOCKLIST.len() == MAX_BLOCKLIST_SIZE {
+        //     return true;
```

Comment:
Also shouldn't we default to `false` here?

Reply:
no, we only want to block what we are certain we want to.  this was an edge case with a full map that made no sense, removed it.